### PR TITLE
Quiet continuous batching at default verbosity

### LINKED ISSUE #48313: Continuous batching is loud by default: WARNING logs and progress bars on a plain generate_batch
A plain `generate_batch` call at default verbosity prints a stream of warnings and a progress bar; `model.generate()` prints nothing. Users embedding CB in a serving loop or trainer get this on every manager start.

## Reproduction

```python
import torch
from transformers import AutoModelForCausalLM, AutoTokenizer, GenerationConfig

model_id = "Qwen/Qwen3-0.6B"
tokenizer = AutoTokenizer.from_pretrained(model_id)
model = AutoModelForCausalLM.from_pretrained(model_id, dtype=torch.bfloat16, device_map="cuda:0")

prompt = tokenizer.apply_chat_template([{"role": "user", "content": "Hi!"}], add_generation_prompt=True)["input_ids"]
model.generate_batch([prompt] * 4, generation_config=GenerationConfig(max_new_tokens=16))
```

## Outcome

```
Solving 4 requests:   0%|          | 0/4 [00:00<?, ?request/s]
ContinuousBatchingLogger - WARNING - `eos_token_id` not set in GenerationConfig. Setting to -1 (disabled).
ContinuousBatchingLogger - WARNING - Continuous batching is much better when using flash attention. Switching from sdpa to flash_attention_3. If you need to use eager or sdpa, use paged|eager or paged|sdpa as the `attn_implementation`.
[transformers] You do not have `flash_attn` installed, using `paged|kernels-community/vllm-flash-attn3` from the `kernels` library instead!
ContinuousBatchingLogger - WARNING - Warming up for continuous batching...
ContinuousBatchingLogger - WARNING - Warming up completed in 8.39s.
Solving 4 requests: 100%|██████████| 4/4 [00:10<00:00,  2.56s/request]
```

(On TP runs there is additionally `Setting NCCL_GRAPH_MIXING_SUPPORT = 0 ...` per rank, and each subsequent manager in the same process repeats the attention-switch and warmup messages.)

Most of these are informational (attention implementation switched, warmup started/finished, NCCL env tweak) and none of them describes a problem with the user's call; at default verbosity they belong at `info`/`debug`. The progress bar should follow the same convention as the rest of the library (opt-in or tied to logging verbosity), since `generate_batch` is used programmatically inside trainers and servers.

## Expected behavior

Silence at default verbosity, matching `model.generate()`. Keep genuinely actionable warnings (e.g. `eos_token_id` unset) at `warning`; move status messages to `info`; gate the `Solving n requests` tqdm behind an argument or the logging level.

## System info

transformers main (a353632), torch 2.13.0+cu130, Python 3.12.12, NVIDIA H100 80GB.

### PR DESCRIPTION
**Pull request by qgallouedec**
[CPU CI](https://transformers-ci.lor-e.huggingface.cool/d/pytest-observability-by-pr/pytest-observability-branch?var-pr=48314) [GPU run-slow](https://transformers-ci.lor-e.huggingface.cool/d/pytest-observability-by-pr/pytest-observability-branch?var-pr=48314)

## What does this PR do?

Fixes #48313. Part of #48294.

Changes:
- The status messages (attention auto-switch, warmup start/finish, progress-bar note) move from `warning` to `info`.
- The `Solving n requests` progress bar becomes opt-in (`progress_bar=False` by default): like `generate`, `generate_batch` prints nothing at default verbosity.
- Actionable warnings stay at `warning`: unset `eos_token_id`, warmup failure, requests dropped during shutdown.

Validation: the issue's repro prints only the `eos_token_id` warning after this change (the repro's GenerationConfig genuinely lacks one; the remaining `[transformers] You do not have flash_attn installed...` line comes from the kernels integration, out of this PR's scope)

## Who can review?

@remi-or (continuous batching)

### CODE REVIEW

File: src/transformers/generation/continuous_batching/continuous_api.py [hunk 1]
```diff
@@ -1217,7 +1217,8 @@ def generate_batch(
             generation_config: Optional generation configuration
             continuous_batching_config: Optional continuous batching configuration
             record_timestamps: If set to true, the requests will have a timestamp for each token generated
-            progress_bar: If set to true, a progress bar will be displayed
+            progress_bar: If set to true, a progress bar will be displayed. Off by default: like `generate`,
+                `generate_batch` prints nothing at default verbosity
```
Comment (qgallouedec) [inline src/transformers/generation/continuous_batching/continuous_api.py]:
```suggestion
            progress_bar: If set to true, a progress bar will be displayed
```

File: src/transformers/generation/continuous_batching/continuous_api.py [hunk 2]
```diff
@@ -650,12 +650,12 @@ def switch_to_cb_friendly_attn(self, model: ProtoPretrainedModel) -> None:
             msg = "Continuous batching is much better when using flash attention."
             if version is not None:
                 target_implem = f"flash_attention_{version}"  # no "paged|" prefix here to enter the branch below
-                logger.warning(
+                logger.info(
```
Comment (remi-or) [inline src/transformers/generation/continuous_batching/continuous_api.py]: I think this is better as a warning, because a user initializing with `sdpa` should k now we change the implementation from under them.

File: src/transformers/generation/continuous_batching/continuous_api.py [hunk 3]
```diff
@@ -1183,10 +1183,10 @@ def continuous_batching_context_manager(
         )
         if warmup and not manager.warmed_up:
             # Warmup is long (~30 sec): best to signal the user it's happening than let them think the manager is stuck
```
Comment (remi-or) [inline src/transformers/generation/continuous_batching/continuous_api.py]:
```suggestion
            # TODO: have a progress bar for the warmup as well, like other inference engine
```
because I agree this might be a bit too verbose but we still need some visibility for the users who want some

**Review by remi-or — CHANGES_REQUESTED** [review]
Some nits but otherwise we are good! Thanks for the improvement

**Review by remi-or — APPROVED** [review]
Some nits but otherwise we are good! Thanks for the improvement

**Comment by qgallouedec** [comment]
Thanks! Applied both in 88e0773c7f.

The two CI failures are unrelated

I'll let you merge this one when you think it's ready
